pallets/connectors/src/lib.rs

```diff
@@ -45,23 +45,52 @@ pub enum ParachainId {
 	Moonbeam,
 }
 
-/// The EVM chain ID
-/// The type should accomodate all chain ids listed on https://chainlist.org/.
-type EVMChainId = u64;
-
 /// A Domain is a chain or network we can send a Connectors message to.
 /// The domain indices need to match those used in the EVM contracts and these
 /// need to pass the Centrifuge domain to send tranche tokens from the other
 /// domain here. Therefore, DO NOT remove or move variants around.
 #[derive(Encode, Decode, Clone, PartialEq, TypeInfo, MaxEncodedLen)]
 #[cfg_attr(feature = "std", derive(Debug))]
 pub enum Domain {
+	/// Referring to the Centrifuge Parachain. Will be used for handling incoming messages.
+	/// NOTE: Connectors messages CAN NOT be sent directly from the Centrifuge chain to the
+	/// Centrifuge chain itself.
+	Centrifuge,
 	/// An EVM domain, identified by its EVM Chain Id
 	EVM(EVMChainId),
-	/// A Polkadot Parachain domain
-	Parachain(ParachainId),
 }
 
+pub(crate) trait ConnectorEncode {
+	fn connector_encode(&self) -> Vec<u8>;
+}
+
+/// Custom encode implementation for the `Domain` type
+/// Domain is encoded as a 9-byte long bytearray, where:
+/// Byte 0 - Flags the Domain variant, i.e, either Domain::Centrifuge (0) or Domain::EVM (1)
+/// Byte 1-9 - Encodes the domain id if applicable;
+///		- For Domain::Centrifuge there's no id, so we append 8 zeros
+///		- For Domain::EVM, encode the respective chain id (8 bytes) as little endian
+///
+/// We need to impl this as a custom encode to not overlap with the `#[derive(Encode, Decode)]`
+/// that are necessary for `Domain` to be used in the storage as a key.
```

Comment:
NOTE: Explained here why I've introduced this trait. I'd be supper happy to receive suggestions on a better way to do this 🙏

Reply:
Thinking about this, if this new encoding is totally representative of the `Domain`, and it does not lose any information, why not use this method as a custom `Encode` and use it also for computing the storage's key?, The "downside" of this is that we need to ensure that now the `Decode` is the inverse of this encoding.

Reply:
> The "downside" of this is that we need to ensure that now the Decode is the inverse of this encoding.

Yeah, this. Plus our custom Encoding implementation is probably not as efficient as the default one which _can_ become an expensive matter when it comes to handling storage keys. Wdyt?

Reply:
mmm... for this type of size, sure it's not a problem. But if we use this pattern for bigger types it could be (?). 
Another approach is to use a Wrapper type with custom encoding. Something like: `DomainSolidity`, which knows how to encode for solidity, and is used only when it requires to be sent to solidity.

Reply:
What advantage do you see in having a `DomainSolidity` wrapper type VS the current `ConnectorEncode` trait? Also, note that the custom encoding is not necessarily "Solidity" dependant, it's the standard format defined for Connectors, thus the `ConnectorsEncode` trait name.

Reply:
> it's the standard format defined for Connectors

Gotcha!

> What advantage do you see in having a DomainSolidity wrapper type VS the current ConnectorEncode trait?

Instead of having two kinds of encodings for a type, you would have two types with the same kind of encoding. I feel this is like "more" type-safety. Depending on where you pass the type, you protect/force the implementation by using the correct encoding call. No mistakes regarding which encoding I should use.

Reply:
Thanks for all the input, @lemunozm. I decided to just implement `Encode` and `Decode` for `Domain` and drop `ConnectorEncode` all together. The custom impls are now covered by tests so we are good :)